src/main/networking/vtunnel.ts

```diff
@@ -0,0 +1,54 @@
+// Maintains configuration for all the processes that go over AF_VSOCK Vtunnel
+
+import fs from 'fs';
+import path from 'path';
+import yaml from 'yaml';
+
+import * as credentialServer from '@/main/credentialServer/httpCredentialHelperServer';
```

Comment:
This is a trivial import loop? Ideally, we should avoid having loops; in this case, it would be better if vtunnel is unaware of the specifics of credential helpers, and has an API to add tunnels or something. (This is basically looking towards how we'd have multiple tunnels later, from unrelated code.)

I'm thinking something like:
```ts
export interface VTunnelConfig {
  name: string;
  handshakePort: number;
  vsockHostPort: number;
  peerAddress: string;
  peerPort: number;
  upstreamServerAddress: string;
};

export class vtunnel {
  // …
  /**
   * Add a new tunnel to the configuration.
   * If a tunnel by that name already exists, it is replaced.
   * @note throws if vtunnel has already started.
   */
  void addTunnel(config VTunnelConfig);
  void start(); // start the background process
};
```
(If possible, make the vtunnel in `background.ts` and pass it in, or alternatively make it a singleton.)

Reply:
I agree with the structure above, TBH the only thing I'm getting out of the CredentialHelper is the port number for the server.